google/cloud/aiplatform/metadata/metadata.py

```diff
@@ -0,0 +1,133 @@
+from google.cloud.aiplatform.metadata.context import _Context
+from google.cloud.aiplatform.metadata.execution import _Execution
+from google.cloud.aiplatform.metadata.artifact import _Artifact
+
+
+class _MetadataService:
+    """Contains the exposed APIs to interact with the Managed Metadata Service."""
+
+    def __init__(self):
+        self._experiment = None
+        self._run = None
+        self._metrics = None
+
+    def set_experiment(self, experiment: str):
+        if not experiment:
+            raise ValueError(f"Invalid experiment {experiment}.")
+
+        store = _MetadataStore.get()
+        if not store:
+            logging.info(
+                f"Creating a default MetadataStore for experiment {experiment}"
+            )
+            _MetadataStore.create()
+
+        context = _Context.get(context_name=experiment)
+        if not context:
+            logging.info(f"Creating a Context for experiment {experiment}")
+            context = _Context.create(
+                context_id=experiment,
+                schema_title="system.Experiment",
+                schema_version="0.0.1",
+            )
+        self._experiment = context.name
+
+    def set_run(self, run: str):
+        if not self._experiment:
+            raise ValueError(
+                "No experiment set for this run. Make sure to call aiplatform.init(experiment='my-experiment') or "
+                "aiplatform.set_experiment(experiment='my-experiment') before trying to set_run. "
+            )
+        if not run:
+            raise ValueError(f"Invalid run {run}.")
+
+        execution = _Execution.get(execution_name=run)
+        if not execution:
+            logging.info(f"Creating an Execution for run {run}")
+            execution = _Execution.create(
+                execution_id=run, schema_title="system.Run", schema_version="0.0.1",
+            )
+        self._run = execution.name
+
+    def log_params(self, params: Dict):
+        if not self._experiment:
+            raise ValueError(
+                "No experiment set for logging parameters. Make sure to call aiplatform.init("
+                "experiment='my-experiment') or aiplatform.set_experiment(experiment='my-experiment') before trying "
+                "to log_params. "
+            )
+        if not self._run:
+            raise ValueError(
+                "No run set for logging parameters. Make sure to call aiplatform.init(experiment='my-experiment', "
+                "run='my-run') or aiplatform.set_run('my-run') before trying to log_params. "
+            )
+        execution = _Execution.get(execution_name=self._run)
+        if not execution:
+            execution = _Execution.create(
+                execution_id=self._run,
+                schema_title="system.Run",
+                schema_version="0.0.1",
+                metadata=params,
+            )
+        else:
+            execution = _Execution.update(execution_name=self._run, metadata=params)
```

Comment:
I believe this would override existing parameters, we would need to append new parameters to existing execution before update.

Reply:
Already changed this to use an instance method. Indeed, we do append new parameters, or new metrics... Tested out and worked. Will share the notebook I used for my experiments shortly